python/graphstorm/distiller/distiller.py

```diff
@@ -0,0 +1,351 @@
+            dataloader_generator,
+            dataset_path=os.path.join(data_dir, 'val'),
+            local_rank=self.rank,
+            world_size=th.distributed.get_world_size(),
+            is_train=False,
+        )
+
+        # get GNN embed dim
+        dataset_iterator = eval_data_provider.get_iterator()
+        if not dataset_iterator:
+            raise RuntimeError("No validation data") 
+        batch = next(iter(dataset_iterator))
+        gnn_embed_dim = batch["labels"].shape[1]
+        self.student.init_proj_layer(gnn_embed_dim=gnn_embed_dim)
+
+        # TODO (HZ): add flexibility to specify different optimizers
+        optimizer = th.optim.Adam(self.student.parameters(), lr=distill_lr)
+        self.student.to(self.device)
+        student = DistributedDataParallel(self.student, device_ids=None if on_cpu else [self.device],
+                                        output_device=None if on_cpu else self.device)
+
+        index = 0
+        global_step = 0
+        while True:
+            th.distributed.barrier()
+            print (f"Train {index + 1}-th shard by trainer {self.rank}")
+            complete, global_step = self.train_shard(
+                global_step=global_step,
+                model=student, 
+                optimizer=optimizer, 
+                train_dataset_provider=train_data_provider, 
+                eval_dataset_provider=eval_data_provider,
+                saved_path=saved_path,
+                save_model_frequency=save_model_frequency,
+                eval_frequency=eval_frequency,
+                max_global_step=max_global_step,
+            )
+            is_first_iteration = False
+
+            index += 1
+            if complete:
+                break
+
+    def save_student_model(self, model, saved_path, global_step):
+        """ Save student model
+
+        Parameters
+        ----------
+        model : GSDistilledModel
+            Distilled student model.
+        saved_path : str
+            Path to save the model.
+        global_step : int
+            Global step of the model checkpoint.
+        """
+        th.distributed.barrier()
+        if self.rank == 0:
+            checkpoint_path = os.path.join(saved_path, f"checkpoint-{global_step}")
+            proj_dir_loc = os.path.join(checkpoint_path, "proj")
+            tokenizer_dir_loc = os.path.join(checkpoint_path, "tokenizer")
+            lm_dir_loc = os.path.join(checkpoint_path, "lm")
+            os.makedirs(proj_dir_loc, exist_ok=True)
+            print (f"Saving checkpoint to {checkpoint_path}")
+
+            # TODO (HZ): need to test if the saved model can be successfully loaded
+            model.module.tokenizer.save_pretrained(tokenizer_dir_loc)
+            model.module.lm.save_pretrained(lm_dir_loc)
+            th.save(model.module.state_dict()["proj"], os.path.join(proj_dir_loc, "pytorch_model.bin"))
+
+        return True
+
+    def to_device(self, inputs, device='cuda'):
```

Comment:
why is this function implemented as a method? it should be implemented as a function in the util script (e.g., python/graphstorm/utils.py).